lib/src/main/java/tech/relaycorp/relaydroid/messaging/ReceiveMessages.kt

```diff
@@ -23,21 +27,15 @@ internal class ReceiveMessages(
         getNonceSigners()
             .flatMapLatest { nonceSigners ->
                 pdcClientBuilder().use {
-                    it.collectParcels(nonceSigners, StreamingMode.CloseUponCompletion)
-                        .mapNotNull { parcelCollection ->
-                            val parcel = try {
-                                parcelCollection.deserializeAndValidateParcel()
-                            } catch (exp: RAMFException) {
-                                logger.log(Level.WARNING, "Malformed incoming parcel", exp)
-                                parcelCollection.ack()
-                                return@mapNotNull null
-                            } catch (exp: InvalidMessageException) {
-                                logger.log(Level.WARNING, "Invalid incoming parcel", exp)
-                                parcelCollection.ack()
-                                return@mapNotNull null
-                            }
-                            IncomingMessage.build(parcel) { parcelCollection.ack() }
-                        }
+                    try {
+                        collectParcels(it, nonceSigners)
+                    } catch (exp: ServerException) {
+                        throw ReceiveMessagesException("Server error", exp)
+                    } catch (exp: ClientBindingException) {
+                        throw ReceiveMessagesException("Client error", exp)
+                    } catch (exp: NonceSignerException) {
+                        throw ReceiveMessagesException("Client signing error", exp)
```

Comment:
I think we should treat this as a `GatewayProtocolException` too: It's technically a protocol violation, it's just that the PoWeb lib checks for it before making a request to the server.